kube_janitor/janitor.py

```diff
@@ -34,15 +35,45 @@ def matches_resource_filter(resource, include_resources: frozenset, exclude_reso
     return resource_included and not resource_excluded and namespace_included and not namespace_excluded
 
 
+def get_expire(resource, ttl_seconds) -> datetime:
+    creation_time = datetime.datetime.strptime(resource.metadata['creationTimestamp'], '%Y-%m-%dT%H:%M:%SZ')
+    return creation_time + datetime.timedelta(seconds=ttl_seconds)
+
+
+def get_delete_notification(expiry_timestamp, delete_notification) -> datetime:
+    return expiry_timestamp - datetime.timedelta(seconds=delete_notification)
+
+
+def add_notification_flag(resource):
+    resource.annotations[NOTIFIED_ANNOTATION] = "yes"
+    resource.update()
+
+
+def was_notified(resource):
+    return NOTIFIED_ANNOTATION in resource.annotations.keys()
+
+
 def get_age(resource):
     creation_time = datetime.datetime.strptime(resource.metadata['creationTimestamp'], '%Y-%m-%dT%H:%M:%SZ')
-    now = datetime.datetime.utcnow()
+    now = utcnow()
     age = now - creation_time
     return age
 
 
+def utcnow():
+    return datetime.datetime.utcnow()
+
+
+def send_delete_notification(resource, reason: str, expire: datetime, dry_run: bool):
+    formatted_expire_datetime = expire.strftime('%Y-%m-%dT%H:%M:%SZ')
+    message = f'{resource.kind} {resource.name} will be deleted at {formatted_expire_datetime} ({reason})'
+    logger.info(message)
+    create_event(resource, message, 'DeleteNotification', dry_run=dry_run)
+    add_notification_flag(resource)
```

Comment:
this should reflect the `dry_run` flag (i.e. do not add any annotation if dry run is set)

Reply:
Make sense. I will update it asap!